src/connector/mysql.rs

```diff
@@ -52,12 +52,30 @@ impl Connectional for Mysql {
     fn with_connection<F, T>(&self, _db: &str, f: F) -> QueryResult<T>
     where
         F: FnOnce(&mut Connection) -> QueryResult<T>,
+        Self: Sized,
     {
         dbg!(self.pool.state());
         let mut conn = self.pool.get()?;
         let result = f(&mut conn);
         result
     }
+
```

Comment:
From here to the end, implement these three only in the trait.

Reply:
@mavilein, why did you add the Sized trait? 
This will, if I'm not mistaken, imply that all concrete implementations will need to be available at compile time.

Reply:
Ah, nevermind, understood. It's better than using `impl Connectional` everywhere.